docs/database-classes.md

```diff
@@ -72,7 +72,76 @@ public abstract class Person
 }
 ```
 
-**Note**: If a database class definition of a Starcounter application contains any non-computed instance properties that are not declared as abstract auto-implemented properties with public get and set accessors and with one of the [supported data types](database-types.md), an exception will be thrown when the application starts.
+**Note**: If a database class definition of a Starcounter application contains any non-computed instance properties that are not declared as abstract auto-implemented properties with public get and set accessors, an exception will be thrown when the application starts.
+
+## Proxy state fields and properties
+
+Sometimes it is required to declare a public non-persistent field or property in a database class. Such fields or properties can be used for advanced calculations or caching of computed values. This can be achieved with usage of the `ProxyStateAttribute`. Starcounter ignores and does not persist everything marked with the `ProxyStateAttribute`.
+
+```csharp
+[Database]
+public abstract class Person
+{
+    public abstract string Name { get; set; }
+    public abstract DateTime CreatedAtUtc { get; set; }
```

Comment:
I think we can remove this property. It doesn't really add anything, and I think we should strive for having samples as small as possible.

Reply:
You mean the `CreatedAtUtc`, right?

```suggestion
```

Reply:
> You mean the `CreatedAtUtc`, right?

Yep.